### PR TITLE
Python3.12 support

### PR DESCRIPTION
**Pull request by jstucke**
- [replaced abandoned package 'si-prefix' with 'quantiphy'](https://github.com/fkie-cad/FACT_core/commit/a547ba17351efd29d8673ada255931c044cda493)
- [updated matplotlib version and no longer pin setuptools version](https://github.com/fkie-cad/FACT_core/commit/c6d16a6bbcf36b3ba08d23f6a9daab22d5fe39af)
- remaining problems:
  - [x] aiohttp is incompatible https://github.com/aio-libs/aiohttp/issues/7639
    - [x] fixed 
    - [x] new release: should be fixed when 3.9 releases
  - [x] uwsgi is incompatible https://github.com/unbit/uwsgi/issues/2566
    - [x] fixed https://github.com/unbit/uwsgi/pull/2542
    - [x] new release
  - [x] yara-python < 4.4.0 is incompatible
    - [x] install new yara and yara-python
    - [x] [update common_analysis_ip_and_uri](https://github.com/fkie-cad/common_analysis_ip_and_uri/pull/7)

### CODE REVIEW

**Comment by maringuu** [comment]
If I'm not mistaken the code in ` src/analysis/plugin/compat.py` must be updated after the yara upgrade.
Also the function is definitely missing a unit test.

**Review by maringuu — COMMENTED** [review]
See above

File: src/analysis/plugin/compat.py [hunk 1]
```diff
@@ -58,19 +58,17 @@ def shutdown(self):
 
 def yara_match_to_dict(match: yara.Match) -> dict:
     """Converts a ``yara.Match`` to the format that :py:class:`analysis.YaraPluginBase` would return."""
-    # FIXME (yara): Use this when we upgrade to yara-python 4.3.0
-    # for string_match in match.strings:
-    #    for string_instance in string_match.instances:
-
-    strings = [(offset, identifier, data.hex()) for offset, identifier, data in match.strings]
+    # see YARA docs: https://yara.readthedocs.io/en/latest/yarapython.html#yara.StringMatchInstance
```
Comment (maringuu) [inline src/analysis/plugin/compat.py:61]: Unnecessary comment?

Comment (jstucke) [reply src/analysis/plugin/compat.py:61]: Hmm I think it is useful since you can't look at the YARA source (because it is a compiled C library)

**Comment by ElDavoo** [comment]
binwalk has not been updated for py3.12 support

**Pull request merged**